packages/otelbin/src/lib/metadataUtils.ts

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2023 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { parse } from "./urlState/jsurl2";
+import JsYaml from "js-yaml";
+import type { IConfig } from "~/components/react-flow/dataType";
+import { type Node } from "reactflow";
+
+export function parseUrlFragment(url: URL) {
+	if (!url) {
+		return {};
+	}
+	const urlHash = url.hash;
+	let parsedConfig = "";
+	if (urlHash != null) {
```

Comment:
**Issue:** This code path is duplication the extraction logic for URL parameters, and it is fragile, e.g., `#distro=otelcol-contrib&config=…` would fail to parse.

Please consider using `toUrlState(toParsedUrlStateParameters(new URLSearchParams(url.hash)), [configBinding])` instead.

Reply:
Fixed and toUrlState is implemented.